src/app/zap_cluster_list.py

```diff
@@ -0,0 +1,146 @@
+    'MEDIA_PLAYBACK_CLUSTER': ['media-playback-server'],
+    'NETWORK_COMMISSIONING_CLUSTER': ['network-commissioning'],
+    'OCCUPANCY_SENSING_CLUSTER': [],
+    'ON_OFF_CLUSTER': ['on-off-server'],
+    'OTA_BOOTLOAD_CLUSTER': [],
+    'POWER_CONFIG_CLUSTER': [],
+    'SCENES_CLUSTER': ['scenes'],
+    'TEMP_MEASUREMENT_CLUSTER': ['temperature-measurement-server'],
+    'WINDOW_COVERING_CLUSTER': [],
+    'ZLL_COMMISSIONING_CLUSTER': []
+}
+
+# List of directories in src/app/clusters to build for client clusters.
+CLIENT_CLUSTERS: typing.Dict[str, typing.List[str]] = {
+    'ALARM_CLUSTER': [],
+    'APPLICATION_BASIC_CLUSTER': [],
+    'BARRIER_CONTROL_CLUSTER': [],
+    'BASIC_CLUSTER': [],
+    'BINDING_CLUSTER': [],
+    'COLOR_CONTROL_CLUSTER': [],
+    'COMMISSIONING_CLUSTER': [],
+    'CONTENT_LAUNCH_CLUSTER': [],
+    'DEVICE_TEMP_CLUSTER': [],
+    'DESCRIPTOR_CLUSTER': [],
+    'DOOR_LOCK_CLUSTER': ['door-lock-client'],
+    'GENERAL_COMMISSIONING_CLUSTER': [],
+    'GROUPS_CLUSTER': ['groups-client'],
+    'GROUP_KEY_MANAGEMENT_CLUSTER': [],
+    'IAS_ZONE_CLUSTER': ['ias-zone-client'],
+    'IDENTIFY_CLUSTER': [],
+    'LEVEL_CONTROL_CLUSTER': [],
+    'LOW_POWER_CLUSTER': [],
+    'MEDIA_PLAYBACK_CLUSTER': ['media-playback-client'],
+    'NETWORK_COMMISSIONING_CLUSTER': [],
+    'OCCUPANCY_SENSING_CLUSTER': [],
+    'ON_OFF_CLUSTER': [],
+    'OTA_BOOTLOAD_CLUSTER': [],
+    'POWER_CONFIG_CLUSTER': [],
+    'SCENES_CLUSTER': ['scenes-client'],
+    'TEMP_MEASUREMENT_CLUSTER': [],
+    'WINDOW_COVERING_CLUSTER': [],
+    'ZLL_COMMISSIONING_CLUSTER': []
+}
+
+
+def get_cluster_sources(clusters: typing.Set[str],
+                        source_map: typing.Dict[str,
+                                                typing.List[str]], side: str):
+    """Returns a list of cluster source directories for the given clusters.
+
+    Returns:
+      The set of source directories to build.
+    """
+
+    cluster_sources: typing.Set[str] = set()
+
+    for cluster in clusters:
+        if not cluster in source_map:
+            raise ValueError("Unhandled %s cluster: %s" % (side, cluster))
+
+        cluster_sources.update(source_map[cluster])
+
+    return cluster_sources
+
+
+def dump_zapfile_clusters(zap_file_path: pathlib.Path):
+    """Prints all of the source directories to build for a given ZAP file.
+
+    Arguments:
+      zap_file_path - Path to the ZAP input file.
+    """
+
+    client_clusters: typing.Set[str] = set()
+    server_clusters: typing.Set[str] = set()
+
+    with open(zap_file_path, "r") as zap_file:
+        zap_json = json.loads(zap_file.read())
+
+        for endpoint_type in zap_json.get('endpointTypes'):
+            for cluster in endpoint_type.get('clusters'):
+                side: str = cluster.get('side')
+                if side == 'client':
+                    clusters_set = client_clusters
+                elif side == 'server':
+                    clusters_set = server_clusters
```

Comment:
I think this is the root issue for clusters that appears in some configuration. The code needs to check for `enabled`.

For example, in the `all-clusters-app.zap`: 
```
{
          "name": "Scenes",
          "code": 5,
          "mfgCode": null,
          "define": "SCENES_CLUSTER",
          "side": "client",
          "enabled": 0,
         ...
}
```

`enabled: 0` is not taken into account, which is probably why `examples/all-clusters-app/all-clusters-common/BUILD.gn` is using those lines: 
```
# These clusters are configured in the ZAP file but don't compile.
   remove_cluster_sources = [
     "ias-zone-client",
     "scenes-client",
   ]
```

Reply:
Done.